### PR TITLE
Docs: Minor SWC docs polish

### PR DESCRIPTION
**Pull request by jonniebigodes**
Follows up on #25379 and #25683

## What I did

With this small pull request, the SWC compiler section was updated to prevent the snippets from not being displayed, as they were removed as part of the mentioned pull request. Additionally the SWC API section was also updated to factor in the feedback from the discussion.

@valentinpalkovic I've assigned this to you as well so we can keep track of it and get the documentation in good shape. Let me know of any feedback you may have, and we'll take it from there. Thanks in advance.

#### The changes in this PR are covered in the following automated tests:
- [ ] stories
- [ ] unit tests
- [ ] integration tests
- [ ] end-to-end tests

#### Manual testing

_This section is mandatory for all contributions. If you believe no manual test is necessary, please state so explicitly. Thanks!_

### Documentation

- [ ] Add or update documentation reflecting your changes
- [ ] If you are deprecating/removing a feature, make sure to update
      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

## Checklist for Maintainers

- [ ] When this PR is ready for testing, make sure to add `ci:normal`, `ci:merged` or `ci:daily` GH label to it to run a specific set of sandboxes. The particular set of sandboxes can be found in `code/lib/cli/src/sandbox-templates.ts`
- [ ] Make sure this PR contains **one** of the labels below:
   <details>
     <summary>Available labels</summary>

     - `bug`: Internal changes that fixes incorrect behavior.
     - `maintenance`: User-facing maintenance tasks.
     - `dependencies`: Upgrading (sometimes downgrading) dependencies.
     - `build`: Internal-facing build tooling & test updates. Will not show up in release changelog.
     - `cleanup`: Minor cleanup style change. Will not show up in release changelog.
     - `documentation`: Documentation **only** changes. Will not show up in release changelog.
     - `feature request`: Introducing a new feature.
     - `BREAKING CHANGE`: Changes that break compatibility in some way with current major version.
     - `other`: Changes that don't fit in the above categories.
   
   </details>

### 🦋 Canary release

This PR does not have a canary release associated. You can request a canary release of this pull request by mentioning the `@storybookjs/core` team here.

_core team members can create a canary release [here](https://github.com/storybookjs/storybook/actions/workflows/canary-release-pr.yml) or locally with `gh workflow run --repo storybookjs/storybook canary-release-pr.yml --field pr=<PR_NUMBER>`_

### CODE REVIEW

File: docs/api/main-config-swc.md [hunk 1]
```diff
@@ -6,7 +6,7 @@ Parent: [main.js|ts configuration](./main-config.md)
 
 Type: `(config: swc.Options, options: Options) => swc.Options | Promise<swc.Options>`
 
-Customize Storybook's [SWC](https://swc.rs/) setup.
+Customize Storybook's [SWC](https://swc.rs/) setup for Webpack-based projects enabled via the [`@storybook/addon-webpack5-compiler-swc`](https://storybook.js.org/addons/@storybook/addon-webpack5-compiler-swc) addon based on the supported [frameworks](../configure/frameworks.md), except Angular.
```
Comment (valentinpalkovic) [inline docs/api/main-config-swc.md:9]: Except Angular, CRA, Ember and Next.js

Comment (jonniebigodes) [reply docs/api/main-config-swc.md]: Appreciate the clarification on this @valentinpalkovic 🙏 ! Updated per latest commit.

File: docs/configure/compilers.md [hunk 2]
```diff
@@ -6,18 +6,7 @@ Javascript compilers are essential in optimizing and transforming code, enhancin
...
-
-<CodeSnippets
-  paths={[
-    'common/main-config-framework-options-builder-use-swc.js.mdx',
-    'common/main-config-framework-options-builder-use-swc.ts.mdx',
-  ]}
-/>
-
-<!-- prettier-ignore-end -->
+SWC is a fast, highly extensible tool for compiling and bundling modern JavaScript applications. Powered by [Rust](https://www.rust-lang.org/), it improves performance and reduces build times. Storybook includes a built-in integration with SWC, allowing zero-configuration setup and built-in types for APIs. If you've initialized Storybook in a Webpack-based project with any of the supported [frameworks](./frameworks.md), except Angular, it will automatically use SWC as its default, providing you with faster loading time.
```
Comment (valentinpalkovic) [inline docs/configure/compilers.md:9]: Same here.

Comment (jonniebigodes) [reply docs/configure/compilers.md]: Done and done, and I appreciate the reminder 🙇 ! I almost forgot about it. I'm glad that we caught it on time.

**Pull request merged**